src/core/lib/gprpp/global_config.h

```diff
@@ -59,6 +59,15 @@
 //
 // Declaring config variables for other modules to access:
 //   GPR_GLOBAL_CONFIG_DECLARE_*TYPE*(name)
+//
+// * Caveat for setting global configs at runtime
+//
+// Setting global configs at runtime multiple times is safe but it doesn't
```

Comment:
This comment does not seem to be in line with the PR description. I'm confused :)
We either read values each time or we don't.

Reply:
Reading the variables multiple times are okay but best practice asks users to do this carefully to use the variable consistently. Also it's useful to change these variables across unit tests but it's not safe with gRPC because grpc_shutdown doesn't guarantee to be called at the end of unit-test.